src/structures/GuildChannel.js

```diff
@@ -265,6 +265,26 @@ class GuildChannel extends Channel {
       .then(() => this);
   }
 
+  /**
+   * Deletes existing permission overwrites for a user or role in this channel.
+   * @param {RoleResolvable|UserResolvable} userOrRole The user or role's permission overwrites to delete
+   * @returns {Promise<GuildChannel>}
+   * @example
+   * // Delete permission overwrites for a message author
+   * message.channel.deleteOverwrite(message.author)
+   *   .then(() => console.log("Deleted permission overwrites for " + message.author.tag + " in " + message.channel))
+   *   .catch(console.error);
+   */
+  deleteOverwrite(userOrRole) {
+    userOrRole = this.guild.roles.resolve(userOrRole) || this.client.users.resolve(userOrRole);
```

Comment:
This will throw for valid but uncached users. Ideally this function should only make an API call and let the API decide whether the role/user exists.

Reply:
Indeed it will, the other methods which this is likely based on do this as well, there is an open discussion about this #5230 
I think if this is being added, in this instance it should definitely be passed directly to discords API

Reply:
I simply followed the way it is currently handled in other methods, although I definitely agree it should use an ID instead of the user or role. Even then I’m not sure if I should change that before your changes get accepted and committed since that would create inconsistencies

Reply:
What I linked is simply a discussion for how we should handle this, it is not a PR yet.

Reply:
Yes, I know, but you proposed changes that weren't accepted yet so it's not worth changing it here since it could create inconsistencies if those changes are never implementes